dbms/src/Flash/Pipeline/Schedule/Tasks/Task.h

```diff
@@ -74,6 +75,18 @@ class Task
     // `TaskHelper::FINALIZE_TASK` can help this.
     void finalize();
 
+    ALWAYS_INLINE void beforeExec(TaskTimer * timer)
+    {
+        assert(nullptr == current_task_timer);
+        current_task_timer = timer;
+        startTraceMemory();
+    }
+    ALWAYS_INLINE static void afterExec()
+    {
+        current_task_timer = nullptr;
+        CurrentMemoryTracker::submitLocalDeltaMemory();
+        current_memory_tracker = nullptr;
```

Comment:
A little strange: in beforeExec, we startTraceMemory, while in afterExec, we don't call endTraceMemory.

Reply:
Good point, I should call `endTraceMemory` instead of set `current_memory_tracker` directly.

Reply:
Done